src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -1424,6 +1424,32 @@ mongoc_uri_parse (mongoc_uri_t *uri, const char *str, bson_error_t *error)
    }
 
    before_slash = scan_to_unichar (str, '/', "", &tmp);
+   if (!before_slash) {
+      // Handle cases of optional delimiting slash
+      char *userpass = NULL;
+      char *hostname = NULL;
```

Comment:
Suggest renaming `hostname` to `hosts` since it may contain multiple comma delimited [Host Identifiers](https://github.com/mongodb/specifications/blob/e651ebe819743f1cffae938f5488c69d5277c21f/source/connection-string/connection-string-spec.md#general-syntax), e.g. `mongodb://host1,host2`.